File: app/models/schemas/analytics.py
from __future__ import annotations

from decimal import Decimal
from typing import Optional

from pydantic import BaseModel


class DashboardSummary(BaseModel):
    total_calls: int
    connected_rate: float
    promise_rate: float
    avg_call_duration: float
    payment_collected: Decimal
    compliance_violations: int


class HourlyBreakdown(BaseModel):
    hour: int
    calls: int
    connected: int
    promise_made: int


class OutcomeDistribution(BaseModel):
    outcome: str
    count: int
    percentage: float


class DashboardResponse(BaseModel):
    summary: DashboardSummary
    hourly_breakdown: list[HourlyBreakdown]
    outcome_distribution: list[OutcomeDistribution]


class CallQualityResponse(BaseModel):
    avg_stt_latency_ms: Optional[float]
    avg_llm_latency_ms: Optional[float]
    avg_tts_latency_ms: Optional[float]
    avg_e2e_latency_ms: Optional[float]
    barge_in_rate: float
    stt_confidence_avg: Optional[float]
    p95_stt_latency_ms: Optional[float]
    p95_llm_latency_ms: Optional[float]
    p95_tts_latency_ms: Optional[float]

File: frontend/src/api/analytics.ts
import client from "./client";

export interface HourlyPoint {
  hour: number;
  calls: number;
  connected: number;
  promise_made: number;
}

export interface OutcomePoint {
  outcome: string;
  count: number;
  percentage: number;
}

export interface DashboardSummary {
  total_calls: number;
  connected_rate: number;
  promise_rate: number;
  avg_call_duration: number;
  payment_collected: string;
  compliance_violations: number;
}

export interface DashboardResponse {
  summary: DashboardSummary;
  hourly_breakdown: HourlyPoint[];
  outcome_distribution: OutcomePoint[];
}

export interface CallQualityResponse {
  avg_stt_latency_ms?: number;
  avg_llm_latency_ms?: number;
  avg_tts_latency_ms?: number;
  avg_e2e_latency_ms?: number;
  barge_in_rate: number;
  stt_confidence_avg: number;
  p95_stt_latency_ms?: number;
  p95_llm_latency_ms?: number;
  p95_tts_latency_ms?: number;
}

export const analyticsApi = {
  getDashboard: async () => {
    const res = await client.get<DashboardResponse>("/analytics/dashboard");
    return res.data;
  },
  getCallQuality: async () => {
    const res = await client.get<CallQualityResponse>("/analytics/call-quality");
    return res.data;
  },
};
